cypress/e2e/GUI/ui-funcionality-tests.cy.js

```diff
@@ -0,0 +1,155 @@
+describe('EngageSphere', () => {
+  beforeEach(() => {
+    Cypress.on('window:before:load', window => {
+      window.document.cookie = 'cookieConsent=accepted'
+    })
+    cy.visit('/');
+  });
+
+  it('Should maintain filters when returning from customer details', () => {
+    cy.get('[data-testid="size-filter"]')
+      .select('Small');
+    cy.contains('button', 'View')
+      .click()
+    cy.get('[class^="CustomerDetail"] h2')
+      .should('be.visible');
+      cy.contains('button', 'Back')
+      .click();
+    cy.get('[data-testid="size-filter"]')
+      .should('have.value', 'Small');
+  });
+
+  it('Should return to the customer list after clicking back', () => {
+    cy.contains('button', 'View')
+      .click()
+    cy.get('[class^="CustomerDetail"] h2')
```

Comment:
Que tal um seletor mais simples, como o seguinte?

```suggestion
    cy.contains('h2', 'Customer Details')
```

Dessa forma, garantimos que o texto correto é exibido no `h2`.